### PR TITLE
fix: changes in path requirements for snakedeploy

### PR DESCRIPTION
**Pull request by AKBrueggemann**
# Description

 Changed the path fetching in common.smk -> get_resource(). Should fetch the total link if snake deploy is used so files can be used from the resources folder in the repository.

## Checklist

- [x] I've formatted the PR title in accordance with the [structure of the conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I've updated the code style using [`pre-commit`](https://ikim-essen.github.io/uncovar/dev-guide/contributing/#pre-commit) if needed.
- [x] I've updated or supplemented the documentation as needed.
- [x] I've read the [`CODE_OF_CONDUCT.md`] document.
- [x] I've read the [`CONTRIBUTING.md`] guide.

[`CODE_OF_CONDUCT.md`]: https://github.com/IKIM-Essen/uncovar/blob/master/CODE_OF_CONDUCT.md
[`CONTRIBUTING.md`]: https://github.com/IKIM-Essen/uncovar/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: workflow/rules/common.smk [hunk 1]
```diff
@@ -212,7 +213,15 @@ def get_fastqs(wildcards):
...
-    return str((Path(workflow.snakefile).parent.parent.parent / "resources") / name)
+    if "https" in str(Path(workflow.snakefile)):
+        dirname = os.path.dirname
+        path_snakefile = str(workflow.snakefile)
+        path_new = dirname(dirname(path_snakefile))
+        folder = "resources"
+        path_complete = f"{path_new}/{folder}/{name}"
+        return path_complete
+    else:
+        return str((Path(workflow.snakefile).parent.parent.parent / "resources") / name)
```
Comment (johanneskoester) [inline workflow/rules/common.smk:224]: This is much simpler, and also more portable as it also works in the cloud:
```suggestion
    return workflow.source_path(f"../../resources/{name}")
```
In addition, I suggest to move the resources subfolder into the workflow subfolder in the future. This makes the separation between resources that come together with the workflow and resources that are downloaded by the workflow cleaner.

**Review by johanneskoester — CHANGES_REQUESTED** [review]
I don't think this will work in all cases (although it might work in your tests now). There is a better solution:
Whenever you use a resource, you should use `workflow.source_path()

**Comment by johanneskoester** [comment]
The error is indeed a bug in Snakemake. Fix is here: https://github.com/snakemake/snakemake/pull/2170

**Comment by johanneskoester** [comment]
Thanks a lot @AKBrueggemann! Once the PR is merged and release, you can rerun the CI in this PR here, as it should automatically take the latest Snakemake.

**Comment by AKBrueggemann** [comment]
Thank you very much @johanneskoester!

**Comment by johanneskoester** [comment]
Needs another fix: https://github.com/snakemake/snakemake/pull/2179

**Pull request merged**